sig-testing/charter.md

```diff
@@ -0,0 +1,142 @@
+  and [kind]
+- Jobs that automate away project toil via [@fejta-bot]
+- Ensuring all of the above is kept running on a best effort basis
+- Tools, frameworks and libraries that make it possible to write tests against
+  kubernetes such as e2e\* or integration test frameworks.
+
+  \* Note that while we are the current de facto owners of the kubernetes e2e
+  test framework, we are not staffed to actively maintain or rewrite it and
+  welcome contributors looking to take on this responsibility.
+
+#### Cross-cutting and Externally Facing Processes
+
+##### Ongoing Support
+
+- The [Release Team test-infra role] is staffed by a member of SIG Testing, as
+  such their responsibilities are within the scope of this SIG, including
+  the maintenance of release jobs
+- We actively collaborate with SIG Contributor Experience, often producing
+  tooling that they are responsible for using to implement polices and
+  processes that they own, e.g. the Github Administration subproject uses
+  [peribolos] and [label_sync] to reduce the toil involved
+- We reserve the right to halt automation and infrastructure that we own,
+  or disable tests that we don't own if the project as a whole is being
+  impacted
+- We are actively assisting with the transition of project infrastructure to
+  the CNCF and enabling non-Googlers to support this
+
+##### Deploying Changes
+
+We aspire to remain agile and deploy quickly, while ensuring a disruption-free
+experience for project contributors. As such, the amount of notice we provide
+and the amount of consensus we seek is driven by our estimation of risk. We
+don't currently define risk in terms of objective metrics, so here is a rough
+description of the guidelines we follow. We anticipate refining these over
+time.
+
+- **Low risk** changes do not break existing contributor workflows, are easy
+  to roll back, and impact at most a few project repos or SIGs. These should
+  be reviewed by another member of SIG Testing or the affected SIG(s),
+  preferably an approver.
+
+- **Medium risk** changes may impact existing contributor workflows, should be
+  easy to roll back, and may impact all of the project's repos. These should
+  be shared with SIG Contributor Experience, may require a lazy consensus
+  issue with [kubernetes-dev@] notice.
+
+- **High risk changes** likely break existing contributor workflows, may be
+  difficult to roll back, and likely impact all of the project's repos. These
+  require a consultation with SIG Contributor Experience, and a lazy consensus
+  issue with [kubernetes-dev@] notice.
+
+### Out of scope
+
+- We are not responsible for troubleshooting or writing tests or jobs for
+  features or subprojects owned by other SIGs
+- We are not responsible for ongoing maintenance of the project's CI Signal,
+  as this is driven by tests and jobs owned by other SIGs. We do however have
+  an interest in producing tools to help improve the signal.
```

Comment:
Do you want to add stuff like on-call are out of scope too?

Is there something you want to add about turn around times for things when they break? (can't guarantee is fine!)

Reply:
I guess I'd rather not get too specific about it if I don't need to.  I stated as "in scope" that we support our infrastructure on a best effort basis.  I aspire for us to have more than that in terms of rigor or metrics, but this is where we are today.

If I have to spell it out, it would be something like:
- As stated above, we support and maintain the project's testing and automation infrastructure on a best effort basis. As such, we do not provide response time guarantees or off-hours support, though we do understand the importance of keeping a global high-traffic open source project moving.

Reply:
let's see if anyone else feels like it should be.